src/app/AttributeCache.cpp

```diff
@@ -219,5 +274,113 @@ CHIP_ERROR AttributeCache::GetStatus(const ConcreteAttributePath & path, StatusI
     return CHIP_NO_ERROR;
 }
 
+void AttributeCache::GetSortedFilters(std::vector<std::pair<DataVersionFilter, size_t>> & aVector)
+{
+    for (auto const & endpointIter : mCache)
+    {
+        EndpointId endpointId = endpointIter.first;
+        for (auto const & clusterIter : endpointIter.second)
+        {
+            if (!clusterIter.second.mCommittedDataVersion.HasValue())
+            {
+                continue;
+            }
+            DataVersion dataVersion = clusterIter.second.mCommittedDataVersion.Value();
+            uint32_t clusterSize    = 0;
+            ClusterId clusterId     = clusterIter.first;
+
+            for (auto const & attributeIter : clusterIter.second.mAttributes)
+            {
+                if (attributeIter.second.Is<StatusIB>())
+                {
+                    clusterSize +=
+                        4; // At least it has control byte +Status + ClusterStatus + end of container = 1 + 1 + 1 + 1 bytes
+                }
+                else
+                {
+                    System::PacketBufferTLVReader bufReader;
+                    bufReader.Init(attributeIter.second.Get<System::PacketBufferHandle>().Retain());
+                    ReturnOnFailure(bufReader.Next());
+                    // Skip to the end of the element.
+                    ReturnOnFailure(bufReader.Skip());
+
+                    // Compute the amount of value data
+                    clusterSize += bufReader.GetLengthRead();
+                }
+            }
+            if (clusterSize == 0)
+            {
+                continue;
+            }
+
+            DataVersionFilter filter(endpointId, clusterId, dataVersion);
+
+            aVector.push_back(std::make_pair(filter, clusterSize));
+        }
+    }
+    std::sort(aVector.begin(), aVector.end(),
+              [](const std::pair<DataVersionFilter, size_t> & x, const std::pair<DataVersionFilter, size_t> & y) {
+                  return x.second > y.second;
+              });
+}
+
+CHIP_ERROR AttributeCache::OnUpdateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                         const Span<AttributePathParams> & aAttributePaths,
+                                                         bool & aEncodedDataVersionList)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    TLV::TLVWriter backup;
+
+    std::vector<std::pair<DataVersionFilter, size_t>> filterVector;
+    GetSortedFilters(filterVector);
+
+    for (auto & filter : filterVector)
```

Comment:
Need to set aEncodedDataVersionList to false up front before we enter this loop.

Reply:
Done, thanks